Source/Core/DolphinQt/Config/SettingsWindow.cpp

```diff
@@ -19,45 +19,134 @@
+      std::pair("palette(alternate-base)", "palette(base)");
+#else
+      std::pair("palette(base)", "palette(alternate-base)");
+#endif
+
+  m_navigation_list->setStyleSheet(
+      QString::fromUtf8(
+          // Remove border around entire widget and adjust background color.
+          "QListWidget { border: 0; background: %1; } "
+          "QListWidget::item { padding: %2; "
+#if defined(__linux__)
+          "} "
+#else
+          // For some reason this is needed to fix inconsistent item padding on Windows and macOS,
+          //  but it unfortunately also breaks item background color changes.
+          "border: 0; } "
+          // Restore selected item color.
+          "QListWidget::item:selected { background: palette(highlight); "
+#if defined(_WIN32)
+          // Prevent text color change on focus loss on Windows.
+          // This seems to breaks the nice white text on macOS.
+          "color: palette(highlighted-text); "
+#endif
+          "} "
+          // Restore hovered item color, though not really the correct color.
+          "QListWidget::item:hover:!selected { background: %3; } "
+#endif
+#if defined(_WIN32)
+          // Remove ugly dotted outline on selected row.
+          "* { outline: none; } "
+#endif
+          )
+          .arg(QString::fromUtf8(list_background))
+          .arg(list_item_padding)
+          .arg(QString::fromUtf8(list_hover_background)));
+
+  layout->addWidget(m_navigation_list);
+
+  auto* const right_side = new QVBoxLayout;
+  layout->addLayout(right_side);
+
+  m_stacked_panes = new QStackedWidget;
+
+  right_side->addWidget(m_stacked_panes);
+
+  // The QFrame gives us some padding around the button.
+  auto* const button_frame = new QFrame;
+  auto* const button_layout = new QGridLayout{button_frame};
+  auto* const button_box = new QDialogButtonBox(QDialogButtonBox::Close);
+  right_side->addWidget(button_frame);
+  button_layout->addWidget(button_box);
 
-  // Add content to layout before dialog buttons.
-  m_tab_widget = new QTabWidget();
-  layout->addWidget(m_tab_widget);
+  connect(button_box, &QDialogButtonBox::rejected, this, &QDialog::reject);
 
-  m_tab_widget->addTab(GetWrappedWidget(new GeneralPane, this, 125, 100), tr("General"));
-  m_tab_widget->addTab(GetWrappedWidget(new InterfacePane, this, 125, 100), tr("Interface"));
-  m_tab_widget->addTab(GetWrappedWidget(new AudioPane, this, 125, 100), tr("Audio"));
-  m_tab_widget->addTab(GetWrappedWidget(new PathPane, this, 125, 100), tr("Paths"));
-  m_tab_widget->addTab(GetWrappedWidget(new GameCubePane, this, 125, 100), tr("GameCube"));
-  m_tab_widget->addTab(GetWrappedWidget(new WiiPane, this, 125, 100), tr("Wii"));
-  m_tab_widget->addTab(GetWrappedWidget(new AdvancedPane, this, 125, 200), tr("Advanced"));
+  connect(m_navigation_list, &QListWidget::currentRowChanged, m_stacked_panes,
+          &QStackedWidget::setCurrentIndex);
+}
+
+SettingsWindow::SettingsWindow(QWidget* parent) : StackedSettingsWindow{parent}
```

Comment:
Is there a reason you didn't group this with the other `SettingsWindow` functions?

Reply:
That's just where it ended up when it was adapted from the old `SettingsWindow`, but I'll move it right now.